modules/apps/portal/portal-mail/src/main/java/com/liferay/portal/mail/sender/internal/messaging/MailSenderMessagingConfigurator.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.mail.sender.internal.messaging;
+
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.messaging.Destination;
+import com.liferay.portal.kernel.messaging.DestinationConfiguration;
+import com.liferay.portal.kernel.messaging.DestinationFactory;
+import com.liferay.portal.kernel.messaging.DestinationNames;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+import com.liferay.portal.mail.sender.internal.configuration.MailSenderMessagingConfiguration;
+
+import java.util.Dictionary;
+import java.util.Map;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Julius Lee
+ */
+@Component(
+	configurationPid = "com.liferay.portal.mail.sender.internal.configuration.MailSenderMessagingConfiguration",
+	immediate = true, service = {}
+)
+public class MailSenderMessagingConfigurator {
+
+	@Activate
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
+
+		_bundleContext = bundleContext;
+
+		MailSenderMessagingConfiguration mailSenderMessagingConfiguration =
+			ConfigurableUtil.createConfigurable(
```

Comment:
use method variable instead of instance variable
https://github.com/liferay-core-infra/liferay-portal/pull/768#discussion_r890994032